### PR TITLE
Add Tanner role

### PR DESCRIPTION
**Pull request by maidxn**

### CODE REVIEW

**Review by nitro2 — DISMISSED** [review]
Please reformat testcases (JSON files), they mix between space and tab.
Add newline at the end of JSON files (and any future file)

**Comment by maidxn** [comment]
May I ask some VScode extension to format these file faster?  (Now I'm just formatting manually each file). Thank you so much

**Review by neihousaigaai — CHANGES_REQUESTED** [review]
Please also fix your test files name,

- Should rename `case-tanner-kill-by-...` to `case-tanner-killed-by-...`
- Fix `testcases/case-tanner-kill-by-wofl-witch-reborn-tanner-win.json` file name

**Review by neihousaigaai — DISMISSED** [review]

**Review by neihousaigaai — APPROVED** [review]

**Review by neihousaigaai — APPROVED** [review]

File: game/__init__.py [hunk 1]
```diff
@@ -620,6 +660,9 @@ async def do_end_daytime_phase(self):
                 voted_user=f"<@{lynched}>", highest_vote_number=votes
             )
 
+            if isinstance(self.players[lynched], roles.Tanner):
+                self.tanner_is_lynced = True
```
Comment (huybt22) [inline game/__init__.py:664]: should be `tanner_is_lynched`

File: game/__init__.py [hunk 2]
```diff
@@ -509,6 +545,10 @@ def get_winning_role(self):
 
         print("DEBUG: ", num_players, num_werewolf)
 
+        # check tanner
+        if self.tanner_is_lynced:
```
Comment (huybt22) [inline game/__init__.py:549]: should be `tanner_is_lynched`

File: game/__init__.py [hunk 3]
```diff
@@ -68,6 +71,7 @@ def reset_game_state(self):
         self.prev_playtime = self.is_in_play_time()
         self.new_moon_mode.set_random_event()
         self.auto_hook = defaultdict(list)
+        self.tanner_is_lynced = False
```
Comment (huybt22) [inline game/__init__.py:74]: should be `tanner_is_lynched`

**Review by huybt22 — CHANGES_REQUESTED** [review]

**Review by huybt22 — DISMISSED** [review]

**Review by neihousaigaai — APPROVED** [review]

**Review by huybt22 — APPROVED** [review]

**Pull request merged**